athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaRecordHandler.java

```diff
@@ -0,0 +1,318 @@
+        super(amazonS3, secretsManager, amazonAthena, SOURCE_TYPE);
+        this.amazonS3 = amazonS3;
+    }
+
+    /**
+     * Used to read the row data associated with the provided Split.
+     *
+     * @param spiller            A BlockSpiller that should be used to write the row data associated with this Split.
+     *                           The BlockSpiller automatically handles chunking the response, encrypting, and spilling to S3.
+     * @param recordsRequest     Details of the read request, including:
+     *                           1. The Split
+     *                           2. The Catalog, Database, and Table the read request is for.
+     *                           3. The filtering predicate (if any)
+     *                           4. The columns required for projection.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing work for a query that has already terminated
+     * @throws IOException
+     */
+    @Override
+    protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recordsRequest, QueryStatusChecker queryStatusChecker)
+            throws IOException {
+        logger.info("readWithConstraint: schema[{}] tableName[{}]", recordsRequest.getSchema(), recordsRequest.getTableName());
+
+        Schema schemaName = recordsRequest.getSchema();
+
+        Split split = recordsRequest.getSplit();
+        String id = split.getProperty("query_id").replace("-","");
+        String fileLocation = split.getProperty("fileLocation");
+        String exportBucket = split.getProperty("exportBucket");
+
+        //get column name and type from the Schema
+        HashMap<String, Types.MinorType> mapOfNamesAndTypes = new HashMap<>();
+        for(Field field : schemaName.getFields()){
+            Types.MinorType minorTypeForArrowType = Types.getMinorTypeForArrowType(field.getType());
+            mapOfNamesAndTypes.put(field.getName(), minorTypeForArrowType);
+        }
+
+        // creating a RowContext class to hold the column name and value.
+        final RowContext rowContext = new RowContext(id);
+
+        //Generating the RowWriter and Extractor
+        GeneratedRowWriter.RowWriterBuilder builder = GeneratedRowWriter.newBuilder(recordsRequest.getConstraints());
+        for (Field next : recordsRequest.getSchema().getFields()) {
+            Extractor extractor = makeExtractor(next, rowContext, mapOfNamesAndTypes);
+            builder.withExtractor(next.getName(), extractor);
+        }
+        GeneratedRowWriter rowWriter = builder.build();
+
+        /*
+         Using S3 Select to read the S3 Parquet files generated in the split
+         */
+
+        //List all the parquet files generated in the split
+        ObjectListing objectListing = amazonS3.listObjects(new ListObjectsRequest().withBucketName(exportBucket).withPrefix(id));
+        for (S3ObjectSummary objectSummary : objectListing.getObjectSummaries()) {
+            if(objectSummary.getSize() > 0){
```

Comment:
if objects are <1 what does it mean?

Reply:
I was trying to say that if the file is not empty (file size > 0), then read it. I was handling a case where Vertica exports an empty file.